eth/backend.go

```diff
@@ -84,17 +82,33 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	b.nonceManager.Unlock(sender)
 
 	data := tx.Data()
-	method, ok := b.methods[string(data[:4])]
-	if !ok {
-		method = "unknown"
+	methodSig := string(data[:4])
+	method, err := b.abiMap[methodSig].MethodById(data[:4])
+	if err != nil {
+		return err
+	}
+	var methodName string
+	if method == nil {
+		methodName = "unknown"
 	}
+	methodName = method.Name
 
-	if err != nil {
-		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, err, strings.Repeat("*", 75))
+	txParams := make(map[string]interface{})
+	if err := decodeTxParams(b.abiMap[methodSig], txParams, data); err != nil {
 		return err
 	}
 
-	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, tx.Hash().String(), strings.Repeat("*", 75))
+	var txParamsString string
+	for _, arg := range method.Inputs {
+		txParamsString += fmt.Sprintf("%v: %v  ", arg.Name, txParams[arg.Name])
+	}
+
+	if sendErr != nil {
+		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"   \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), methodName, strings.TrimSpace(txParamsString), sendErr, strings.Repeat("*", 75))
+		return sendErr
+	}
+
+	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\". Inputs: \"%v\"  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), methodName, txParamsString, tx.Hash().String(), strings.Repeat("*", 75))
```

Comment:
Missing a call to `strings.TrimSpace()` for `txParamsString` here. If we trim the whitespace for `txParamsString` once prior to these log statements then we can avoid having to remember to call `strings.TrimSpace` in two different places.

Reply:
fixed in b47e031